### PR TITLE
test: pkg/client/ocm unit tests

### PR DESCRIPTION
**Pull request by pawelpaszki**
## Description
test: pkg/client/ocm unit tests

Note - the client itself can't be tested, because mock for connection cannot be created, as connection is not an interface. Hence  `pkg/client/ocm/client.go` is ignored in the tests coverage

## Checklist (Definition of Done)

- [x] All acceptance criteria specified in JIRA have been completed
- [x] Unit and integration tests added that prove the fix is effective or the feature works (tested against emulated and non-emulated OCM environment)
- ~~[ ] Documentation added for the feature~~
- [x] CI and all relevant tests are passing
- [ ] Code Review completed
- [ ] Verified independently by reviewer
- [ ] All PR comments are resolved either by addressing them or creating follow up tasks
- ~~[ ] Required metrics/dashboards/alerts have been added (or PR created).~~
- ~~[ ] Required Standard Operating Procedure (SOP) is added.~~
- ~~[ ] JIRA has been created for changes required on the client side~~

### CODE REVIEW

File: pkg/client/ocm/types_test.go [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+	}{
+		{
+			name: "should return RHOSAK for standard quota",
+			fields: fields{
+				t: StandardQuota,
+			},
+			want: string(RHOSAKProduct),
+		},
+		{
+			name: "should return RHOSAKTrial for standard quota",
```
Comment (ziccardi) [inline pkg/client/ocm/types_test.go:27]:
```suggestion
			name: "should return RHOSAKTrial for developer quota",
```

File: pkg/client/ocm/types_test.go [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+	}{
+		{
+			name: "should return 'rhosak' for standard quota",
+			fields: fields{
+				t: StandardQuota,
+			},
+			want: ResourceName,
+		},
+		{
+			name: "should return 'rhosak' for standard quota",
```
Comment (ziccardi) [inline pkg/client/ocm/types_test.go:62]:
```suggestion
			name: "should return 'rhosak' for developer quota",
```

Comment (pawelpaszki) [reply pkg/client/ocm/types_test.go]: good catch. thanks

File: pkg/client/ocm/types_test.go (hunk 1 above)
Comment (pawelpaszki) [reply pkg/client/ocm/types_test.go]: good catch. thanks

**Comment by wtrocki** [comment]
> Note - the client itself can't be tested, because mock for connection cannot be created, as connection is not an interface

Maybe https://github.com/matryer/moq would help in that matter?
I have been using it for such situations (interface is good to have but not required)

**Comment by pawelpaszki** [comment]
> > Note - the client itself can't be tested, because mock for connection cannot be created, as connection is not an interface
> 
> Maybe https://github.com/matryer/moq would help in that matter? I have been using it for such situations (interface is good to have but not required)

we have been using it as well, but unless I am wrong, you can't mock a struct (which the mentioned `Connection` used by the client is). If you think that such mock can be created, I would love to see how this can be obtained

File: pkg/client/ocm/config_test.go [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+			},
+			wantErr: false,
+		},
+	}
+
+	RegisterTestingT(t)
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			config := tt.fields.config
```
Comment (ziccardi) [inline pkg/client/ocm/config_test.go:53]: config is always `NewOCMConfig()`.
We can remove it from the fields and change this line to:
```suggestion
			config := NewOCMConfig()
```

File: pkg/client/ocm/id_test.go [hunk 4]
```diff
@@ -28,18 +30,14 @@ func Test_idGenerator_Generate(t *testing.T) {
...
-			}
-			got := i.Generate()
-			if err := tt.validateFn(got); err != nil {
-				t.Errorf("Generate() = %v", err.Error())
-			}
-			if len(got) > MaxClusterNameLength {
-				t.Errorf("Generated ID length should not exceed 15 chars: %v", got)
-			}
+			got := NewIDGenerator(tt.fields.prefix).Generate()
+			Expect(tt.validateFn(got)).To(BeNil())
```
Comment (ziccardi) [inline pkg/client/ocm/id_test.go:39]:
```suggestion
			Expect(tt.validateFn(got)).To(Succeed())
```

File: pkg/client/ocm/id_test.go [hunk 5]
```diff
@@ -28,18 +30,14 @@ func Test_idGenerator_Generate(t *testing.T) {
...
+			Expect(len(got) > MaxClusterNameLength).To(BeFalse())
```
Comment (ziccardi) [inline pkg/client/ocm/id_test.go:40]:
```suggestion
			Expect(len(got)).To(BeNumerically("<=", MaxClusterNameLength))
```

File: pkg/client/ocm/types_test.go [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+			},
+			want: string(RHOSAKTrialProduct),
+		},
+	}
+
+	RegisterTestingT(t)
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			Expect(tt.fields.t.GetProduct()).To(Equal(tt.want))
```
Comment (ziccardi) [inline pkg/client/ocm/types_test.go:39]: [optional]
You can declare `want` as `KafkaProduct`, so you can avoid the castings to string.
Then you just have to change the test to:
```suggestion
			Expect(tt.fields.t.GetProduct()).To(BeEquivalentTo(tt.want))
```

**Review by ziccardi — APPROVED** [review]
lgtm

**Pull request merged**